pkg/flow/internal/controller/node_custom_component.go

```diff
@@ -54,6 +58,29 @@ type CustomComponentNode struct {
 
 var _ ComponentNode = (*CustomComponentNode)(nil)
 
+// ExtractImportAndDeclare extracts an importNamespace and a customComponentName from a componentName.
+// TODO: this function assumes that CustomComponentNames and ImportNamespaces don't contain "."
+// In other words, there are two possible scenarios:
+// - [customComponentName] LABEL ->  instance of a local declare
+// - [importNamespace].[customComponentName] LABEL -> instance of an imported declare
+// To address this limitation in the future, we will need to find a different approach to
+// identify the importNamespaces and CustomComponentNames.
+func ExtractImportAndDeclare(componentName string) (string, string) {
+	parts := strings.SplitN(componentName, ".", 2)
+	if len(parts) == 0 {
+		return "", ""
+	}
+	// If this is a local declare.
+	importNamespace := ""
+	customComponentName := parts[0]
+	// If this is an imported declare.
+	if len(parts) > 1 {
+		importNamespace = parts[0]
+		customComponentName = parts[1]
+	}
+	return importNamespace, customComponentName
+}
```

Comment:
In cases where a function returns two values of the same type, I'd recommend either naming the return values so it's obvious to callers which value is which or creating a struct which names the values. 

Here's an example of the first case (also introducing a switch statement to maybe make things more clear):

```go
func ExtractImportAndDeclare(componentName string) (importNamespace, customComponentName string) {
  parts := strings.SplitN(componentName, ".", 2)
  switch len(parts) {
  case 1: // [customComponentName] 
    customComponentName = parts[0] 
  case 2: // [importNamespace].[customComponentName] 
    importNamespace = parts[0]
    customComponentName = parts[1] 
  }

  return importNamespace, customComponentName 
}
```

Reply:
Nice, I did not know this was possible in Go, that works really well in this case, thanks